### DISCUSSION TITLE
Mouse support?

### DISCUSSION THREAD

**Discussion by henrysheehy** [sourcelair/xterm.js#4957] [General]
How can I enable mouse input? Specifically, I want to easily click text in Vim to move the mouse pointer and highlight for copying. Thank you.

**Comment by jerch** [comment]
@henrysheehy You might want to read about mouse reports in xterm to get an idea how applications can use mouse reports (link: https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-Mouse-Tracking). The supported encodings and protocols you will find in https://github.com/xtermjs/xterm.js/blob/master/src/common/services/CoreMouseService.ts, before inventing something very custom you might want to check first, if your mouse ideas fit any of these to stay compatible what most terminal emulators can do.

If the defaults above don't suit your ideas, you will have to build your own mouse event handlers as you'd do for any other web app.
